### DISCUSSION TITLE
Limit navigatin sidebar depth

### DISCUSSION THREAD

**Discussion by DominikMa** [pradyunsg/furo#321] [Ideas] [answered]
Thank you for the wonderful theme!

I tried to limit the depth of the navigation sidebar with `:maxdepth:` or `:tocdepth:` but neither worked.

Adding a python API reference using `autosummary::` generates a nice overview page/table with the documented modules/classes listed. Listing these again in the sidebar seems unnecessary to me. If a module has a lot of sub modules/classes the sidebar will be filled with a unreadable and chaotic list.

For example in the PyTorch [docs](https://pytorch.org/docs/stable/nn.html)  in the `torch.nn` module there are a bunch of classes. These classes appear in the  `autosummary::` table, but not in the sidebar.

I tried to get something similar.

**Comment by DominikMa** [comment, marked as answer]
Just saw that you already answered that on https://github.com/pradyunsg/furo/discussions/289. So the question is answered.
I still think this is relevant for API references.

**Comment by Eric-Arellano** [comment]
Hey @pradyunsg, would you be interested in a PR to add an `html_theme_option` for `navigation_depth`? We strongly suspect based on experiments & your comment in https://github.com/pradyunsg/furo/discussions/227#discussioncomment-1249750 that the Qiskit docs are taking 3.5 hours to build due to how many API docs we have, compared to 0.5 hours on Pytorch:

> The entire site structure is presented in the sidebar -- if you have a big documentation set, all your pages get correspondingly large as well.

Turns out, it was really easy to let the user limit the max depth with `html_options = {"navigation_depth": 2}`. Default to `-1`. The PR would be about 5 lines to your `__init__.py` and `theme.conf`:

https://github.com/Qiskit/qiskit_sphinx_theme/pull/451/files#diff-162292e53ecb07a82b73eb2b51f19288e62d09f7b6978233501c4b07f38f70f2

**Reply by pradyunsg** [reply]
Adding a x-ref to https://github.com/pradyunsg/furo/discussions/289#discussioncomment-1584287
